src/hotspot/share/oops/instanceKlassMiscStatus.cpp

```diff
@@ -57,4 +58,13 @@ void InstanceKlassMiscStatus::assign_class_loader_type(const ClassLoaderData* cl
     set_shared_class_loader_type(ClassLoader::APP_LOADER);
   }
 }
+
+#ifdef ASSERT
+void InstanceKlassMiscStatus::assert_is_safe(bool set) {
+  // Setting a flag is safe if it's set once or at a safepoint. RedefineClasses can set or
```

Comment:
Is this a generally true statement, or only true for specific flags?

Reply:
For InstanceKlass, it's true.  There are flags that are set at runtime outside a safepoint that require atomic bit set (is_being_redefined for example) that are in AccessFlags.  They would fail this test.  The flags in MiscStatus::_flags pass this test.